### PR TITLE
feat: add auth0 provider

### PR DESCRIPTION
**Pull request by shahargl**

### CODE REVIEW

File: examples/alerts/new_auth0_users.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Alert when there are new GitHub Stars utilizing keepstate
```
Comment (talboren) [inline examples/alerts/new_auth0_users.yml:1]:
```suggestion
# Alert when there are new Auth0 users
```

File: keep/providers/auth0_provider/auth0_provider.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        )
+
+    def notify(self, **kwargs):
+        pass  # Define how to notify about any alerts or issues
+
+    def dispose(self):
+        pass
+
+
+class Auth0LogsProvider(Auth0Provider):
```
Comment (talboren) [inline keep/providers/auth0_provider/auth0_provider.py:63]: Doesn't really seem to let you query logs but rather more specific to new users?

File: keep/providers/auth0_provider/auth0_provider.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+
+if __name__ == "__main__":
+    import logging
+
+    logging.basicConfig(level=logging.DEBUG, handlers=[logging.StreamHandler()])
+
+    # If you want to use application default credentials, you can omit the authentication config
+    config = {
+        # "authentication": {"credentials_path": "/path/to/your/service_account.json"},
```
Comment (talboren) [inline keep/providers/auth0_provider/auth0_provider.py:133]: ?

File: keep/providers/auth0_provider/auth0_provider.py (hunk 2 above)
Comment (shahargl) [reply keep/providers/auth0_provider/auth0_provider.py:63]: It does (you query by log type)

**Pull request merged**